pkg/cover/report.go

```diff
@@ -61,28 +63,28 @@ func (rg *ReportGenerator) Do(w io.Writer, pcs []uint64) error {
 	for i, pc := range pcs {
 		pcs[i] = PreviousInstructionPC(rg.arch, pc)
 	}
-	covered, prefix, err := rg.symbolize(pcs)
+	covered, err := rg.symbolize(pcs)
 	if err != nil {
 		return err
 	}
 	if len(covered) == 0 {
 		return fmt.Errorf("'%s' does not have debug info (set CONFIG_DEBUG_INFO=y)", rg.vmlinux)
 	}
 	uncoveredPCs := rg.uncoveredPcsInFuncs(pcs)
-	uncovered, prefix2, err := rg.symbolize(uncoveredPCs)
+	uncovered, err := rg.symbolize(uncoveredPCs)
 	if err != nil {
 		return err
 	}
-	if len(uncoveredPCs) != 0 {
-		prefix = combinePrefix(prefix, prefix2)
-	}
-	return rg.generate(w, prefix, covered, uncovered)
+	return rg.generate(w, covered, uncovered)
 }
 
-func (rg *ReportGenerator) generate(w io.Writer, prefix string, covered, uncovered []symbolizer.Frame) error {
+func (rg *ReportGenerator) generate(w io.Writer, covered, uncovered []symbolizer.Frame) error {
 	var d templateData
 	for f, covered := range fileSet(covered, uncovered) {
-		remain := filepath.Clean(strings.TrimPrefix(f, prefix))
+		if strings.HasPrefix(f, rg.buildDir) {
+			return fmt.Errorf("path '%s' doesn't match build dir '%s'", f, rg.buildDir)
+		}
+		remain := filepath.Clean(strings.TrimPrefix(f, rg.buildDir))
 		if rg.srcDir != "" && !strings.HasPrefix(remain, rg.srcDir) {
```

Comment:
This condition now should always be true, right?
Both KernelSrc and KernelBuildSrc are not empty, we always remove KernelBuildSrc, so we should always add KernelSrc.